3rdParty/iresearch/core/utils/async_utils.cpp

```diff
@@ -179,21 +178,20 @@ void read_write_mutex::unlock(bool exclusive_only /*= false*/) {
   // if have write lock
   if (owns_write()) {
     if (exclusive_owner_recursion_count_) {
-      if (!exclusive_only) { // a recursively locked mutex is alway top-level write locked
+      if (!exclusive_only) { // a recursively locked mutex is always top-level write locked
         --exclusive_owner_recursion_count_; // write recursion unlock one level
       }
 
       return;
     }
 
     ADOPT_SCOPED_LOCK_NAMED(mutex_, lock);
-    static std::thread::id unowned;
 
     if (exclusive_only) {
-      ++concurrent_count_; // aquire the read-lock
+      ++concurrent_count_; // acquire the read-lock
     }
 
-    exclusive_owner_.store(unowned);
+    exclusive_owner_.store(std::thread::id());
```

Comment:
This thread is currently unlocking the mutex, so why are we storing our thread-id as owner here (again)?

Reply:
No, this is not storing the current thread's id.
This is the constructor of `std::thread::id()`, which, according to [cppreference](https://en.cppreference.com/w/cpp/thread/thread/id/id) 
> Default-constructs a new thread identifier. The identifier does not represent a thread.

Reply:
🤦‍♂ wer lesen kann ist klar im Vorteil... 🙄